src/RiotPls.DataDragon/Converters/ChampionTypeJsonConverter.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using RiotPls.DataDragon.Enums;
+
+namespace RiotPls.DataDragon.Converters
+{
+    public sealed class ChampionTypeJsonConverter : JsonConverter<ChampionType>
+    {
+        private static readonly ChampionType[] _values = (ChampionType[])Enum.GetValues(typeof(ChampionType));
+        public override ChampionType Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var result = default(ChampionType);
+
+            if (reader.TokenType == JsonTokenType.StartArray)
+            {
+                reader.Read();
+
+                while (reader.TokenType != JsonTokenType.EndArray)
+                {
+                    result |= Enum.Parse<ChampionType>(reader.GetString());
+                    reader.Read();
+                }
+            }
+            else
+                throw new Exception($"Api breaking change: Expected to deserialize an array but got: {reader.TokenType}");
```

Comment:
I think we should make a more specific Exception, waiting for suggestions.

I'm kinda thinking on RiotApiException? or ApiBreakingChangeException... I'm unsure.

Reply:
Yes we should have our own `RiotApiException` with an kind of "type", or something depending on the status code?

Reply:
Probably. Also the status code wouldn't be needed because of inner exceptions, which would be an HttpException which already includes the HttpCode

Reply:
Nvm, if it's in a converter we could let it be an Exception for now.